python/pyspark/pandas/frame.py

```diff
@@ -1454,11 +1462,196 @@ def corr(self, method: str = "pearson") -> "DataFrame":
         There are behavior differences between pandas-on-Spark and pandas.
 
         * the `method` argument only accepts 'pearson', 'spearman'
-        * the data should not contain NaNs. pandas-on-Spark will return an error.
-        * pandas-on-Spark doesn't support the following argument(s).
+        * if the `method` is `spearman`, the data should not contain NaNs.
+        * if the `method` is `spearman`, `min_periods` argument is not supported.
+        """
+        if method not in ["pearson", "spearman", "kendall"]:
+            raise ValueError(f"Invalid method {method}")
+        if method == "kendall":
+            raise NotImplementedError("method doesn't support kendall for now")
+        if min_periods is not None and not isinstance(min_periods, int):
+            raise TypeError(f"Invalid min_periods type {type(min_periods).__name__}")
```

Comment:
Seems like pandas allows float:
```python
>>> pdf.corr('pearson', min_periods=1.4)
          dogs      cats
dogs  1.000000 -0.851064
cats -0.851064  1.000000
```
But I'm not sure if it's intended behavior or not, since they raises `TypeError: an integer is required` when the type is `str` as below:

```python
>>> pdf.corr('pearson', min_periods='a')
Traceback (most recent call last):
...
TypeError: an integer is required
>>>
```

Reply:
I am also not sure, but the type of `min_periods` is also expected to be `int` in Pandas.
I think that `pdf.corr('pearson', min_periods=1.4)` can work in Pandas just because a validation is missing in Pandas